packages/blade/src/components/Input/_decisions/decisions.md

```diff
@@ -0,0 +1,212 @@
+  showClearButton
+  helpText="Enter first name and last name. Eg: Kamlesh Chandnani"
+  errorText="Name cannot be blank"
+  successText="Name validated"
+  validationState={inputValue.length < 10 ? 'error' : 'success'}
+/>
+```
+
+The mobile keyboard type will be automatically determined based on the `type` attribute. Check out [Web & React Native Attributes for showing keyboard types and autocomplete suggestions based on input](#web--react-native-attributes-for-showing-keyboard-types-and-autocomplete-suggestions-based-on-input) section for more details
+
+| Prop                                                                                               | Type                                                                      | Required                                                                            | Default                                                   | Description                                                                                                                                                                                                                                                                           |
+| -------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------- | ----------------------------------------------------------------------------------- | --------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| type                                                                                               | `text`, `search`, `tel`, `email`, `url`, `numeric`                        | No                                                                                  | `text`                                                    | type of text to be rendered. this will also determine the keyboard type to be show on mobile devices                                                                                                                                                                                  |
+| label                                                                                              | `string`                                                                  | (confirm if we have a use case for input field without labels(search field maybe?)) |                                                           | Determines the label of an input field. Also used as `aria-label`                                                                                                                                                                                                                     |
```

Comment:
Can we do some typescript magic to make `ariaLabel` property required when `label` is not given? like one of `label` or `ariaLabel` is required 🤔